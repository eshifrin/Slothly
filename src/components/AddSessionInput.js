import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput } from 'react-native';
import PropTypes from 'prop-types';
import db from '../db/firebase';

export default class AddSessionInput extends Component {
  state = { sessionLength: '' };

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          style={styles.input}
          keyboardType="number-pad"
          placeholder="HOW MANY MINUTES DIDYA MEDITATE?"
          returnKeyType="done"
          style={styles.input}
          value={this.state.sessionLength}
          onChangeText={sessionLength => this.setState({ sessionLength })}
          onSubmitEditing={this.addSession}
        />
      </View>
    );
  }

  addSession = () => {
    const sessionLength = Number(this.state.sessionLength);

    if (sessionLength) {
      this.props.addSession(sessionLength);
      this.setState({ sessionLength: '' });
    }
  };
}

AddSessionInput.propTypes = {
  addSession: PropTypes.func
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    borderColor: 'black',
    borderWidth: 1
  },
  input: {
    flex: 1,
    height: 60,
    textAlign: 'center',
    fontFamily: 'futura',
    fontSize: 16
  }
});
