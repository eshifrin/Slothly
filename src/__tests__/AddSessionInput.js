import 'react-native';
import React from 'react';
import AddSessionInput from '../components/AddSessionInput';
import { shallow } from 'enzyme';

describe('AddSessionInput', () => {
  let wrapper;
  const mockAddSession = jest.fn();

  beforeEach(done => {
    wrapper = shallow(<AddSessionInput addSession={mockAddSession} />);
    setTimeout(() => done(), 0);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls addSession with correct parameters', () => {
    const mockSessionLength = '35';
    wrapper.setState({ sessionLength: mockSessionLength });
    wrapper.instance().addSession();
    expect(mockAddSession).toHaveBeenCalled;
    expect(mockAddSession).toHaveBeenCalledWith(Number(mockSessionLength));
  });
});
