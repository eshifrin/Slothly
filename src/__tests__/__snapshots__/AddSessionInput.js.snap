// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSessionInput renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "paddingHorizontal": 16,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="HOW MANY MINUTES DIDYA MEDITATE?"
    returnKeyType="done"
    style={
      Object {
        "flex": 1,
        "fontFamily": "futura",
        "fontSize": 16,
        "height": 60,
        "textAlign": "center",
      }
    }
    value=""
  />
</View>
`;
